how-to/register-with-browser/client/src/provider.ts

```diff
@@ -146,10 +146,87 @@ function getCustomActions(): CustomActionsMap {
 					});
 				}
 			}
+		},
+		announce: async (payload: CustomActionPayload): Promise<void> => {
+			if (payload.callerType === CustomActionCallerType.CustomButton) {
+				console.info("Announce called with payload:", payload);
+				await showPopup(
+					{ width: 400, height: 300 },
+					payload.windowIdentity,
+					"Announce",
+					"Announce the application to anyone listening ?",
+					[
+						{
+							id: "yes",
+							label: "Yes",
+							default: true
+						},
+						{
+							id: "no",
+							label: "No"
+						}
+					]
+				);
+			}
 		}
 	};
 }
 
+async function showPopup(
+	dimensions: { width: number; height: number },
+	parentIdentity: OpenFin.Identity,
+	title: string,
+	instructions: string,
+	buttons: {
+		id: string;
+		label: string;
+		default?: boolean;
+	}[]
+): Promise<string | undefined> {
+	console.log("Parent Identity", parentIdentity);
+
+	const browserWindow = fin.Window.wrapSync(parentIdentity);
+	const parentBounds = await browserWindow.getBounds();
+
+	console.log("Parent Bounds", parentBounds);
+
+	const halfParentWidth = parentBounds.width / 2;
+	const halfParentHeight = parentBounds.height / 2;
+	const halfWidth = dimensions.width / 2;
+	const halfHeight = dimensions.height / 2;
+
+	const parentCenter = {
+		x: parentBounds.left + halfParentWidth,
```

Comment:
This should not be including parent window bounds. The x, y coordinates used below in the call to showPopupWindow already take into account the window left/top.

Reply:
Confirmed working, thanks @gallak-openfin